scheduler/util.go

```diff
@@ -728,7 +728,9 @@ func inplaceUpdate(ctx Context, eval *structs.Evaluation, job *structs.Job,
 			Tasks:          option.TaskResources,
 			TaskLifecycles: option.TaskLifecycles,
 			Shared: structs.AllocatedSharedResources{
-				DiskMB: int64(update.TaskGroup.EphemeralDisk.SizeMB),
+				DiskMB:   int64(update.TaskGroup.EphemeralDisk.SizeMB),
+				Ports:    update.Alloc.AllocatedResources.Shared.Ports,
+				Networks: update.Alloc.AllocatedResources.Shared.Networks,
```

Comment:
@nickethier pining if you happen to know, but noticed networks was dropped too, I'm guessing we want to carry over all fields

Reply:
I think so, but I think we'd want a copy of it. I wonder if we should just set `Shared` on line 730 to `update.Alloc.AllocatedResources.Shared.Copy()`?

Reply:
Yeah, its not clear to me why `DiskMB` is `update.TaskGroup.EphemeralDisk.SizeMB` instead of `update.Alloc.AllocatedResources.Shared.DiskMB`

Reply:
Ah I see, above Ports and Networks are set from the existing alloc because those cannot be updated inplace and is guarded by `tasksUpdated`. 
The same is happening when setting the `DiskMB` here. So I think we can leave as is but would still add a `.Copy()` to the `Networks`.